### PR TITLE
Convert Visualization Settings Actions to TS

### PR DESCRIPTION
**Pull request by oisincoveney**
Converts the visualization actions to typescript. There's only two so it's relatively simple as a refactor. 

I also took the liberty to rename some of the dashcard function names to reflect that they are using the function made for dashboards, not the one for the query builder. Those names confused me for a little bit.

### CODE REVIEW

File: frontend/src/metabase/query_builder/actions/visualization-settings.ts [hunk 1]
```diff
@@ -57,8 +63,3 @@ export const replaceAllCardVisualizationSettings =
       }),
     );
   };
-
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/actions/visualization-settings.ts:60]: I don't think we need these, all usages seem to have been replaced

**Review by heypoom — COMMENTED** [review]
Nice! The new types and namings looks good. I think `onUpdateVisualizationSettings` is still used by a click action though.

File: frontend/src/metabase/query_builder/actions/visualization-settings.ts (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/query_builder/actions/visualization-settings.ts:60]: huh. I think I must have had a bad search then.

Comment (oisincoveney) [reply frontend/src/metabase/query_builder/actions/visualization-settings.ts:60]: Wait @heypoom I changed the name of the original function to `onUpdateVisualizationSettings` - we should be checking for any references that still refer to `updateCardVisualizationSettings` or `replaceAllCardVisualizationSettings`

File: frontend/src/metabase/query_builder/actions/visualization-settings.ts [hunk 2]
```diff
@@ -9,9 +12,11 @@ import {
 
 import { updateQuestion } from "./core";
 
-export const updateCardVisualizationSettings =
-  settings => async (dispatch, getState) => {
+export const onUpdateVisualizationSettings =
```
Comment (oisincoveney) [inline frontend/src/metabase/query_builder/actions/visualization-settings.ts:15]: @heypoom over here

**Review by heypoom — APPROVED** [review]

File: frontend/src/metabase/query_builder/actions/visualization-settings.ts (hunk 1 above)
Comment (oisincoveney) [reply frontend/src/metabase/query_builder/actions/visualization-settings.ts:60]: no worries 😄 thanks for checking!

**Review by WiNloSt — APPROVED** [review]
looks good to me

**Pull request merged**